tests/MongoDB.Driver.Examples/DocumentationExamples.cs

```diff
@@ -1568,6 +1568,46 @@ public void Index_Example_2()
             result.Should().Be("cuisine_1_name_1");
         }
 
+        [Fact]
+        public void Causal_Consistency_Example_2()
+        {
+            RequireServer.Check().SupportsCausalConsistency();
+
+            this.client.GetDatabase("myDatabase").DropCollection("myCollection");
+
+            // Start Tunable consistency controls example
+            var connectionString = "mongodb://localhost/?readPreference=secondaryPreferred";
+
+            var client = new MongoClient(connectionString);
+            var database = client.GetDatabase("myDatabase");
+            var collection = database.GetCollection<BsonDocument>("myCollection");
+
+            var document = new BsonDocument("name", "Tom");
+
+            // Start client session, which is causally consistent by default
+            using (var session = client.StartSession(new ClientSessionOptions()))
+            {
+                //Run causally related operations within the session
+                collection.InsertOne(session, document);
+                collection.UpdateOne(session,
+                    Builders<BsonDocument>.Filter.Eq(bsonDocument => bsonDocument["name"], "Tom"),
+                    Builders<BsonDocument>.Update.Set(bsonDocument => bsonDocument["name"], "John"));
```

Comment:
I would only use lambdas with POCOs, not with BsonDocument. These lines could just be:

    // Run causally related operations within the session
    collection.InsertOne(session, new BsonDocument("name", "Tom"));
    collection.UpdateOne(session,
        Builders<BsonDocument>.Filter.Eq("name", "Tom"),
        Builders<BsonDocument>.Update.Set("name", "John"));